yt/data_objects/region_expression.py

```diff
@@ -136,8 +136,23 @@ def _create_slice(self, slice_tuple):
             # We'll do this by getting the x/y axes, and then using that.
             width = source.right_edge[xax] - source.left_edge[xax]
             height = source.right_edge[yax] - source.left_edge[yax]
-            # Make a resolution tuple with
+            # We have two cases here.  The first is that the imaginary component
+            # is *positive*.  In that case, the component is the number of
+            # values to sample.
+            # In the case that it is *negative*, the absolute value of it
+            # corresponds to the factor of the base resolution of the dataset in
+            # that dimension.
             resolution = (int(new_slice[xax].step.imag), int(new_slice[yax].step.imag))
+            if resolution[0] < 0:
+                resolution = (
+                    self.ds.domain_dimensions[xax] * -resolution[0],
+                    resolution[1],
+                )
+            if resolution[1] < 0:
+                resolution = (
+                    resolution[0],
+                    self.ds.domain_dimensions[yax] * -resolution[1],
+                )
             sl = sl.to_frb(width=width, resolution=resolution, height=height)
```

Comment:
Modifying a tuple's content is a little awkward. I think it would flow much more naturally if we avoided making a tuple too early.
```suggestion
            if xres := int(new_slice[xax].step.imag) < 0:
                xres *= - self.ds.domain_dimensions[xax]
            if yres := int(new_slice[yax].step.imag) < 0:
                yres *= - self.ds.domain_dimensions[yax]
            sl = sl.to_frb(width=width, resolution=(xres, yres), height=height)
```